src/main/java/buildcraftAdditions/items/ItemCanister.java

```diff
@@ -67,16 +71,9 @@ public void registerIcons(IIconRegister register) {
 		overlay = RenderUtils.registerIcon(register, "fluidOverlay");
 	}
 
-	@Override
-	@SideOnly(Side.CLIENT)
-	public IIcon getIconFromDamage(int damage) {
```

Comment:
When you do a 'setTextureName()' MC will automatically load the icon and set itemIcon to this and it will properly return the icon on getIcon.